Fix crash when disconnecting before connecting.

clydebarrow opened the pull request:
Fixes #107.

sarahlensing commented on lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImpl.java:
```diff
@@ -68,11 +70,12 @@
+    connectState = ConnectState.IDLE;
   }
 
   public void connect(Context context, LostApiClient.ConnectionCallbacks callbacks) {
```
When `connect` is called we should update the `connectState` to `ConnectState.CONNECTING`

clydebarrow commented:
I've added a test for that specific case, and refined the handling of `connectState`, building on your suggested change. It's now possible for the state to go from IDLE to CONNECTING back to IDLE cleanly. Also fixed test failures revealed as a consequence. All tests now pass.

ecgreb commented on lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImpl.java:
```diff
@@ -19,6 +10,15 @@
 import android.os.Looper;
 import android.util.Log;
 
+import com.mapzen.android.lost.api.FusedLocationProviderApi;
```
Library imports above android imports

ecgreb commented on lost/src/main/java/com/mapzen/android/lost/internal/LostApiClientImpl.java:
```diff
@@ -18,7 +18,8 @@ public LostApiClientImpl(Context context, ConnectionCallbacks callbacks) {
     this.connectionCallbacks = callbacks;
   }
 
-  @Override public void connect() {
```
Inline method annotations preferred

ecgreb commented on lost/src/main/java/com/mapzen/android/lost/internal/LostApiClientImpl.java:
```diff
@@ -1,10 +1,10 @@
 import com.mapzen.android.lost.api.LocationServices;
 import com.mapzen.android.lost.api.LostApiClient;
 
-import android.content.Context;
```
Library imports above android imports

ecgreb commented on lost/src/main/java/com/mapzen/android/lost/internal/LostApiClientImpl.java:
```diff
@@ -28,20 +29,22 @@ public LostApiClientImpl(Context context, ConnectionCallbacks callbacks) {
     clientManager.addClient(this);
   }
 
-  @Override public void disconnect() {
```
Inline method annotations preferred

ecgreb commented on lost/src/main/java/com/mapzen/android/lost/internal/LostApiClientImpl.java:
```diff
@@ -52,7 +55,8 @@ public LostApiClientImpl(Context context, ConnectionCallbacks callbacks) {
     getFusedLocationProviderApiImpl().disconnect();
   }
 
-  @Override public boolean isConnected() {
```
Inline method annotations preferred

ecgreb commented on lost/src/test/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImplTest.java:
```diff
@@ -1,5 +1,11 @@
 package com.mapzen.android.lost.internal;
 
+import android.app.PendingIntent;
```
Library imports above android imports
